src/ts/core/features/filter-search.ts

```diff
@@ -0,0 +1,130 @@
+import {Feature, Settings} from '../settings'
+import {browser} from 'webextension-polyfill-ts'
+import {waitForSelectorToExist} from 'src/core/common/mutation-observer'
+
+export const config: Feature = {
+    id: 'filter_search',
+    name: 'Search pages in page or references filter',
+    enabledByDefault: true,
+}
+
+const checkSettingsAndSetupFilterSearchToggle = () => {
+    Settings.isActive('filter_search').then(active => {
+        toggleFilterSearch(active)
+    })
+}
+checkSettingsAndSetupFilterSearchToggle()
+
+browser.runtime.onMessage.addListener(async message => {
+    if (message === 'settings-updated') {
+        checkSettingsAndSetupFilterSearchToggle()
+    }
+})
+
+const toggleFilterSearch = (active: boolean) => {
+    if (active) {
+        document.addEventListener('click', filterSearch)
+    } else {
+        document.removeEventListener('click', filterSearch)
+    }
+}
+
+const searchInputId = 'searchInputBox'
```

Comment:
nit: but maybe prefix like `roam-toolkit--searchInputBox`

Clarifies it's not a native roam thing, and stands out to people who might want to customize the css.

Reply:
I'm curious why two dashes? :)

Reply:
Hmm, I guess it's a convention for denoting hierarchy from http://getbem.com/introduction/ , although the meaning isn't quite the same here